[OperationalInsights] Support for new commands and update existing

dabenhamMic opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

<!-- Please add a brief description of the changes made in this PR -->

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

msJinLei commented:
I notice you add new cmdlets in your PR and so please submit a cmdlet review here https://github.com/Azure/azure-powershell-cmdlet-review-pr

Before submit the review, please refer to the [cmdlet design guideline] (https://github.com/Azure/azure-powershell/tree/master/documentation/development-docs/design-guidelines) first.

In static analysis check, break changes are detected in your codes. Please refer to https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md and avoid breaking change. We can discuss in cmdlet review in details.

You may want to add some tests or record existing test cases. Please refer to the https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/azure-powershell-developer-guide.md for more details.

dabenhamMic commented:
@msJinLei added new cmdlets request issue:  https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/951

dabenhamMic commented:
Hi @msJinLei,
@yoramsinger asked me to stop updating and expanding OperationalInsights PSH module. 
I have to utilize the existing commands and update their contracts.
what is needed to approve and merge this PR?

BethanyZhou commented:
Hi @dabenhamMic , some revisions need be done in cmdlet design doc, please have a look.

dabenhamMic commented:
@BethanyZhou what else is missing here?

BethanyZhou commented:
@dabenhamMic , we need finish your design review before code implementation, please have a look: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/951

isra-fel commented:
Since the design review is ongoing let me make this a draft PR. Please mark it as "Ready for review" when you feel comfortable. Thanks

dabenhamMic commented:
> Since the design review is still ongoing let me make this a draft PR, for repo maintenance purposes. Please mark it as "Ready for review" when you feel comfortable. Thanks

I have updated the open issue: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/951 
added a summary of all new commands and changes

dabenhamMic commented:
Hey @isra-fel  , I have gone over the errors in the build - there are some stuff that I have fixed (wrong types and other issues that I have resolved) how can we disable some of the errors? please ping me in Teams (My Microsoft alias is: dabenham)

dabenhamMic commented:
**Remaining errors:**
  "Az.OperationalInsights","Microsoft.Azure.Commands.OperationalInsights.Clusters.GetAzureOperationalInsightsClusterCommand","Get-AzOperationalInsightsCluster","0","1050","The parameter set '**__AllParameterSets**' for cmdlet 'Get-AzOperationalInsightsCluster' has been removed.","Add parameter set '__AllParameterSets' back to cmdlet 'Get-AzOperationalInsightsCluster'."
    **=> false positive**
  "Az.OperationalInsights","Microsoft.Azure.Commands.OperationalInsights.SetAzureOperationalInsightsStorageInsightCommand","Set-AzOperationalInsightsStorageInsight","0","1050","The parameter set 'ByWorkspaceObject' for cmdlet 'Set-AzOperationalInsightsStorageInsight' has been removed.","Add parameter set 'ByWorkspaceObject' back to cmdlet 'Set-AzOperationalInsightsStorageInsight'."
    **=> nothing have changed - ByWorkspaceObject still in use in this command**
  "Az.OperationalInsights","Microsoft.Azure.Commands.OperationalInsights.SetAzureOperationalInsightsStorageInsightCommand","Set-AzOperationalInsightsStorageInsight","0","1050","The parameter set 'ByWorkspaceName' for cmdlet 'Set-AzOperationalInsightsStorageInsight' has been removed.","Add parameter set 'ByWorkspaceName' back to cmdlet 'Set-AzOperationalInsightsStorageInsight'."
    **=> nothing have changed - ByWorkspaceName still in use in this command**
  "Az.OperationalInsights","Microsoft.Azure.Commands.OperationalInsights.SetAzureOperationalInsightsStorageInsightCommand","Set-AzOperationalInsightsStorageInsight","0","1050","The parameter set '**__AllParameterSets**' for cmdlet 'Set-AzOperationalInsightsStorageInsight' has been removed.","Add parameter set '__AllParameterSets' back to cmdlet 'Set-AzOperationalInsightsStorageInsight'."
    **=> false positive**
  "Az.OperationalInsights","Microsoft.Azure.Commands.OperationalInsights.Clusters.UpdateAzureOperationalInsightsClusterCommand","Update-AzOperationalInsightsCluster","0","1020","The cmdlet 'Update-AzOperationalInsightsCluster' no longer has output type 'Microsoft.Azure.Commands.OperationalInsights.Models.PSLinkedService'.","Make cmdlet 'Update-AzOperationalInsightsCluster' return type 'Microsoft.Azure.Commands.OperationalInsights.Models.PSLinkedService'."
    **=>this was a mistake - now fixed - false alarm**

  "Az.OperationalInsights","Microsoft.Azure.Commands.OperationalInsights.Clusters.UpdateAzureOperationalInsightsClusterCommand","Update-AzOperationalInsightsCluster","0","1050","The parameter set '**__AllParameterSets**' for cmdlet 'Update-AzOperationalInsightsCluster' has been removed.","Add parameter set '__AllParameterSets' back to cmdlet 'Update-AzOperationalInsightsCluster'."
    **=> false positive**

VeryEarly commented on src/OperationalInsights/OperationalInsights/Client/OperationalInsightsClient.LinkedServices.cs:
```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.Azure.Commands.OperationalInsights.Models;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/Client/OperationalInsightsClient.LinkedStorageAccount.cs:
```diff
@@ -0,0 +1,104 @@
+﻿using Microsoft.Azure.Commands.OperationalInsights.Models;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/Client/OperationalInsightsClient.Operations.cs:
```diff
@@ -0,0 +1,16 @@
+﻿using Microsoft.Azure.Commands.OperationalInsights.Models;
```
please add license

dabenhamMic replied:
added

VeryEarly commented on src/OperationalInsights/OperationalInsights/Client/OperationalInsightsClient.Purge.cs:
```diff
@@ -0,0 +1,36 @@
+﻿using System.Management.Automation;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/Client/OperationalInsightsClient.Tables.cs:
```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.Azure.Commands.OperationalInsights.Models;
```
please add license

dabenhamMic replied:
added

VeryEarly commented on src/OperationalInsights/OperationalInsights/Client/OperationalInsightsClient.Workspaces.cs:
```diff
@@ -148,12 +161,15 @@ public virtual PSWorkspace UpdatePSWorkspace(UpdatePSWorkspaceParameters paramet
                 parameters.WorkspaceName,
                 workspace.Location,
-                string.IsNullOrWhiteSpace(parameters.Sku) ? workspace.Sku : parameters.Sku,
+                parameters?.Sku == null ? new PSWorkspaceSku(workspace.Sku, workspace.CapacityReservationLevel) : parameters.Sku,
```
how can parameters be null?

VeryEarly commented on src/OperationalInsights/OperationalInsights/DataExports/GetAzureOperationalInsightsDataExportCommand.cs:
```diff
@@ -0,0 +1,47 @@
+﻿using Microsoft.Azure.Commands.OperationalInsights.Client;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/DataExports/NewAzureOperationalInsightsDataExportCommand.cs:
```diff
@@ -0,0 +1,58 @@
+﻿using Microsoft.Azure.Commands.OperationalInsights.Models;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/DataExports/RemoveAzureOperationalInsightsDataExportCommand.cs:
```diff
@@ -0,0 +1,62 @@
+﻿using Microsoft.Azure.Commands.OperationalInsights.Models;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/DataExports/UpdateAzureOperationalInsightsDataExportCommand.cs:
```diff
@@ -0,0 +1,93 @@
+﻿using Microsoft.Azure.Commands.OperationalInsights.Models;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/Models/CreatePSDataExportParameters.cs:
```diff
@@ -0,0 +1,15 @@
+﻿using System;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/Models/PSAvailableServiceTiers.cs:
```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.Azure.Management.OperationalInsights.Models;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/Models/PSClusterSku.cs:
```diff
@@ -36,19 +41,24 @@ public PSClusterSku(ClusterSku sku)
+            if (this.Capacity < 1000)
             {
-                throw new PSArgumentException("SkuCapacity need to be in range '1000 - 2000' ");
+                throw new PSArgumentException("SkuCapacity need to be above 1000 GB ");
```
```suggestion
                throw new PSArgumentException("SkuCapacity need to be more than1000 GB ");
```

VeryEarly commented on src/OperationalInsights/OperationalInsights/Models/PSDataExport.cs:
```diff
@@ -0,0 +1,61 @@
+﻿using Microsoft.Azure.Management.OperationalInsights.Models;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/Models/PSOperation.cs:
```diff
@@ -0,0 +1,32 @@
+﻿using System;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/Models/PSOperationStatus.cs:
```diff
@@ -0,0 +1,27 @@
+﻿using System;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/Models/PSSavedSearchParameters.cs:
```diff
@@ -0,0 +1,93 @@
+﻿using Microsoft.Azure.Management.OperationalInsights.Models;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/Models/PSStorageInsight.cs:
```diff
@@ -35,15 +36,21 @@ public PSStorageInsight(StorageInsight storageInsight, string resourceGroupName,
-            this.Name = storageInsight.Name;
-            this.ResourceId = storageInsight.Id;
 
             if (storageInsight != null)
```
why is this if block needed?

dabenhamMic replied:
removed

VeryEarly commented on src/OperationalInsights/OperationalInsights/Models/PSTable.cs:
```diff
@@ -0,0 +1,38 @@
+﻿using Microsoft.Azure.Management.OperationalInsights.Models;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/Models/PSWorkspaceFeatures.cs:
```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.Azure.Management.OperationalInsights.Models;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/Models/PSWorkspaceSku.cs:
```diff
@@ -0,0 +1,76 @@
+﻿using System;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/Models/UpdatePSTableParameters.cs:
```diff
@@ -0,0 +1,14 @@
+﻿using System;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/Models/WorkspacePurge/PSWorkspacePurgeBodyFilters.cs:
```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.Azure.Management.OperationalInsights.Models;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/Models/WorkspacePurge/PSWorkspacePurgeResponse.cs:
```diff
@@ -0,0 +1,18 @@
+﻿using Microsoft.Azure.Management.OperationalInsights.Models;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/Models/WorkspacePurge/PSWorkspacePurgeStatusResponse.cs:
```diff
@@ -0,0 +1,17 @@
+﻿using Microsoft.Azure.Management.OperationalInsights.Models;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/OperationStatus/GetAzureOperationalInsightsOperationStatusCommand.cs:
```diff
@@ -0,0 +1,29 @@
+﻿using Microsoft.Azure.Commands.OperationalInsights.Models;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/OperationalInsights.csproj:
```diff
@@ -1,17 +1,13 @@
 
   <Import Project="$(MSBuildThisFileDirectory)..\..\Az.props" />
 
-  <PropertyGroup>
```
any reason removing this section?

VeryEarly commented on src/OperationalInsights/OperationalInsights/Operations/GetAzureOperationalInsightsOperationCommand.cs:
```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections.Generic;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/Operations/GetAzureOperationalInsightsOperationCommand.cs:
```diff
@@ -0,0 +1,16 @@
+
+namespace Microsoft.Azure.Commands.OperationalInsights.Operations
+{
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "OperationalInsightsOperation", SupportsShouldProcess = true), OutputType(typeof(IList<PSOperation>))]
```
```suggestion
    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "OperationalInsightsOperation"), OutputType(typeof(IList<PSOperation>))]
```

VeryEarly commented on src/OperationalInsights/OperationalInsights/Tables/GetAzureOperationalInsightsTableCommand.cs:
```diff
@@ -0,0 +1,38 @@
+﻿using System;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/Tables/SetAzureOperationalInsightsTableCommand.cs:
```diff
@@ -0,0 +1,49 @@
+﻿using Microsoft.Azure.Commands.OperationalInsights.Models;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/WorkspacePurge/GetAzureOperationalInsightsWorkspacePurgeStatusCommand.cs:
```diff
@@ -0,0 +1,34 @@
+﻿using Microsoft.Azure.Commands.OperationalInsights.Models;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/WorkspacePurge/NewAzureOperationalInsightsPurgeWorkspaceCommand.cs:
```diff
@@ -0,0 +1,75 @@
+﻿using Microsoft.Azure.Commands.OperationalInsights.Models;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/Workspaces/UpdateAzureOperationalInsightsWorkspaceSharedKeysCommand.cs:
```diff
@@ -0,0 +1,30 @@
+﻿using Microsoft.Azure.Commands.OperationalInsights.Models;
```
please add license

VeryEarly commented on src/OperationalInsights/OperationalInsights/AvailableServiceTiers/GetAzureOperationalInsightsAvailableServiceTiersCommand.cs:
```diff
@@ -0,0 +1,26 @@
+﻿using Microsoft.Azure.Commands.OperationalInsights.Models;
```
please add license

dabenhamMic replied:
what do you mean here?

VeryEarly commented:
test cases need to be re-recorded after API upgrade.

dabenhamMic commented on src/OperationalInsights/OperationalInsights/Az.OperationalInsights.psd1:
```diff
@@ -12,7 +12,7 @@
 
 # Version number of this module.
-ModuleVersion = '2.3.0'
+ModuleVersion = '2.3.1'
```
change undone

dabenhamMic commented on src/OperationalInsights/OperationalInsights/Az.OperationalInsights.psd1:
```diff
@@ -90,40 +90,52 @@ CmdletsToExport = 'New-AzOperationalInsightsAzureActivityLogDataSource',
                'Get-AzOperationalInsightsSavedSearch', 
                'Get-AzOperationalInsightsSchema', 
-               'New-AzOperationalInsightsComputerGroup', 
+               'New-AzOperationalInsightsComputerGroup', # editted - return contract
```
removed

dabenhamMic commented on src/OperationalInsights/OperationalInsights/Client/OperationalInsightsClient.AvailableServiceTiers.cs:
```diff
@@ -0,0 +1,29 @@
+﻿using Microsoft.Azure.Commands.OperationalInsights.Models;
```
what do you mean? what is the license?

dabenhamMic replied:
added

dabenhamMic commented on src/OperationalInsights/OperationalInsights/Client/OperationalInsightsClient.DataExports.cs:
```diff
@@ -0,0 +1,113 @@
+﻿using Microsoft.Azure.Commands.OperationalInsights.Models;
```
added

dabenhamMic commented:
adding @yoramsinger  - Yabo asked us to re-recorded the api calls.
this will take some more time (I haven't done this before) and will hold us back, please see if you can ask for an exception for us?

dabenhamMic commented:
@VeryEarly  can we proceed with this pr?

wyunchi-ms commented:
/azp run

wyunchi-ms commented:
This PR will introduce breaking changes. Please keep it open until the major release.

dingmeng-xue commented on src/OperationalInsights/OperationalInsights/ChangeLog.md:
```diff
@@ -18,8 +18,53 @@
+  - `Get-AzOperationalInsightsPurgeWorkspaceStatus`
+
+
+## Version 2.4.0
```
```suggestion
## Version 2.3.1
```

wyunchi-ms commented on src/OperationalInsights/generated/utils/Unprotect-SecureString.ps1:
```diff
@@ -0,0 +1,16 @@
+try {
+    $plaintext = [System.Runtime.InteropServices.Marshal]::PtrToStringBSTR($ssPtr)
+} finally {
+    [System.Runtime.InteropServices.Marshal]::ZeroFreeBSTR($ssPtr)
```
Please remove the files under `generated` folder.

wyunchi-ms commented:
You can skip the api check in the other modules.

dabenhamMic reviewed (commented):
already fixed

wyunchi-ms commented:
/azp run azure-powershell - security-tools

wyunchi-ms commented:
LGTM

isra-fel commented on src/OperationalInsights/OperationalInsights/LinkedStorageAccount/SetAzureOperationalInsightsLinkedStorageAccountCommand.cs:
```diff
@@ -38,9 +38,9 @@ public class SetAzureOperationalInsightsLinkedStorageAccountCommand : Operationa
+            HelpMessage = "Data Source Type should be one of 'CustomLogs', 'AzureWatson', 'Query', 'Alerts'.")]
         [ValidateNotNullOrEmpty]
-        [ValidateSet("CustomLogs", "AzureWatson")]
+        [ValidateSet("CustomLogs", "AzureWatson", "Query", "Alerts", IgnoreCase = true)]
```
I don't know how frequently the values of DataSourceType change, but if you don't want to block customers to use your latest type, you can change `ValidateSet` to `ArgumentCompleter` which provides the same tab-completion feature but does not validate the argument.

dabenhamMic replied:
these are ARM resources types, it can be left as is

isra-fel commented on src/OperationalInsights/OperationalInsights/Operations/GetAzureOperationalInsightsOperationCommand.cs:
```diff
@@ -0,0 +1,28 @@
+
+namespace Microsoft.Azure.Commands.OperationalInsights.Operations
+{
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "OperationalInsightsOperation"), OutputType(typeof(IList<PSOperation>))]
```
```suggestion
    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "OperationalInsightsOperation"), OutputType(typeof(PSOperation))]
```
The cmdlet does not return a list, instead, it returns the operation for many times, so the output type should be operation

dabenhamMic replied:
@isra-fel  is there a way to list all cmdlets that have "breaking changes"?

isra-fel replied:
We have limited types of breaking change. Do a full-text search of [these attributes](https://github.com/Azure/azure-powershell/blob/main/documentation/breaking-changes/breaking-changes-definition.md#cmdlets) you should be able to list all of them

isra-fel reviewed (changes requested):
Hello Danial @dabenhamMic , please see my inline comments.
Besides, since a bunch of breaking changes were introduced, could you help us fill in a table like this so we can provide our customer the migration guide? (Do this after the PR is merged so that everything is finalized).
Thanks.

|Cmdlet| Breaking Change| Old Usage| New Usage|
|-|-|-|-|
|| | ||

dabenhamMic commented:
Hi @isra-fel  , I have addressed all issues mentioned in your last review

isra-fel commented on src/OperationalInsights/OperationalInsights/LinkedStorageAccount/NewAzureOperationalInsightsLinkedStorageAccountCommand.cs:
```diff
@@ -36,11 +37,12 @@ public class NewAzureOperationalInsightsLinkedStorageAccountCommand : Operationa
+        [ValidateSet("CustomLogs", "AzureWatson", "Query", "Alerts", IgnoreCase = true)]
         public string DataSourceType { get; set; }
 
+        [CmdletParameterBreakingChange("StorageAccountId")]
```
```suggestion
```
Since renaming this parameter to plural might not be a good idea, we might just remove this line completely, as there is not going to be breaking change.

isra-fel reviewed (changes requested):
Others look good to me :)

dabenhamMic commented:
Hey @wyunchi-ms , @isra-fel 
could you approve and merge this pr?

wyunchi-ms commented:
/azp run azure-powershell - security-tools

isra-fel reviewed (approved):
LGTM

isra-fel commented:
@dabenhamMic Hey Daniel, the PR is good now and I'll merge it once security-tools check finishes. As I said, there's one more thing we need for the release, which is the migration guide. Would you kindly fill in a table like this and  send it to us?

Thanks.

Cmdlet | Breaking Change | Old Usage | New Usage
-- | -- | -- | --

dabenhamMic commented:
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/dabenham/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/dabenham/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>
<!--table
	{mso-displayed-decimal-separator:"\.";
	mso-displayed-thousand-separator:"\,";}
@page
	{margin:.75in .7in .75in .7in;
	mso-header-margin:.3in;
	mso-footer-margin:.3in;}
tr
	{mso-height-source:auto;}
col
	{mso-width-source:auto;}
br
	{mso-data-placement:same-cell;}
td
	{padding-top:1px;
	padding-right:1px;
	padding-left:1px;
	mso-ignore:padding;
	color:black;
	font-size:11.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:Calibri, sans-serif;
	mso-font-charset:0;
	mso-number-format:General;
	text-align:general;
	vertical-align:bottom;
	border:none;
	mso-background-source:auto;
	mso-pattern:auto;
	mso-protection:locked visible;
	white-space:nowrap;
	mso-rotate:0;}
-->
</style>
</head>

<body link="#0563C1" vlink="#954F72">



Cmdlet | Breaking Change | Old Usage | New Usage
-- | -- | -- | --
Get-AzOperationalInsightsCluster | The parameter set '__AllParameterSets' for cmdlet   'Get-AzOperationalInsightsCluster' has been removed | no default parameter set | default parameter set is now "list", when providing RG name -   return all clusters for the given RG. Added support for input cluster   resourceID
Update-AzOperationalInsightsCluster | The cmdlet 'Update-AzOperationalInsightsCluster' no longer has output   type 'Microsoft.Azure.Commands.OperationalInsights.Models.PSLinkedService | no cahnge - bug fix |  
Update-AzOperationalInsightsCluster | __AllParameterSets' for cmdlet 'Update-AzOperationalInsightsCluster' has   been removed. |   | now supports ResourceID (instead of manually entering RG name and cluster   name parameters), added support for input cluster object for update



</body>

</html>

isra-fel commented:
Great, thanks Daniel!
